src/main/java/Duke.java

```diff
@@ -1,10 +1,431 @@
+        String taskType = tasks.get(inputTaskIndex - 1).getTaskType();
+        String taskMark = "";
+        switch (taskType) {
+        case "ToDo":
+            taskMark = "[T]";
+            break;
+
+        case "Deadline":
+            taskMark = "[D]";
+            break;
+
+        case "Event":
+            taskMark = "[E]";
+            break;
+
+        default:
+            break;
+        }
+        return taskMark;
+    }
+
+    public static void processInput(){   
+        Scanner scan = new Scanner ( System.in ); 
+        String inData = scan.nextLine(); // user line of input
+        String[] inLine = inData.split(" "); // code process line of input into an array
+        
+        boolean isFirstLine = false; 
+
+        while (!inData.equals("bye")){
+            if (isFirstLine) { // kick start the loop if first time
+                inData = scan.nextLine();
+                inLine = inData.split(" ");
+            } else {
+                isFirstLine = true;
+            }
+
+            int inputTaskIndex = Integer.parseInt(inLine[1]);
+            int taskIndex = inputTaskIndex - 1;
+            switch (inLine[0]){
+            case "list":
+                printList(tasks);
+                break;
+
+            case "mark":
+            try {
+                if (inputTaskIndex > tasks.size() || inputTaskIndex < 1) {
+                    throw new InvalidListIndexException();
+                }
+
+                tasks.get(taskIndex).setTaskDone(true);
+                
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
+
+                System.out.println("Nice! I've marked this task as done: \n" + taskMark + doneMark +
+                        tasks.get(taskIndex).getTaskTitle());
+
+                String newEntry = parseTaskToString(tasks);
+                writeDisk(newEntry);
+            } catch (InvalidListIndexException | NumberFormatException e) {
+                System.out.println("Error - List index given invalid. Please check again.");
+            } catch (IOException e) {
+                System.out.println("Error when saving file: " + e.getMessage());
+            }
+            break;
+            
+            case "unmark":
+            try {
+                if (inputTaskIndex > tasks.size() || inputTaskIndex < 1) {
+                    throw new InvalidListIndexException();
+                }
+
+                tasks.get(taskIndex).setTaskDone(false);
+
+
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
```

Comment:
Similarly, in such cases, you could consider replacing the output with constant variables to eliminate magic literals.